### ISSUE TITLE
Zero-length string is valid for supported aria attributes

### ISSUE THREAD

**Issue by carmacleod** [validator/validator#1027]
The last paragraph of section [8.6 State and Property Attribute Processing](https://www.w3.org/TR/wai-aria-1.2/#state_property_processing) in the ARIA spec says:
> Sometimes states and properties are present in the DOM but have a zero-length string ("") as their value. Authors MAY specify a zero-length string ("") for any supported (but not required) state or property. User agents SHOULD treat state and property attributes with a value of "" the same as they treat an absent attribute.

This means that, for example, `aria-activedescendant=""` is valid on supporting roles, which are listed in the "Used in Roles" and "Inherits into Roles" rows in [aria-activedescendant's Characteristics table](https://w3c.github.io/aria/#aria-activedescendant).

See also: https://github.com/validator/validator/issues/557#issuecomment-687471513

**Comment by sideshowbarker** [comment]
For the purposes of implementing support for that *“Authors MAY specify a zero-length string ("") for any supported (but not required) state or property”* requirement in the HTML checker, can I safely assume that what the ARIA editors and the ARIA working group actually meant is something like this:

> Authors MAY specify a zero-length string ("") for any supported (but not required) state or property, *and MAY specify any any supported (but not required) state or property with just the attribute name without any value at all*.

If I can actually assume that, then I’d be curious to know why the ARIA editors and the ARIA working group didn’t choose wording like that to begin with — rather than using the current wording they chose, which seems to only allow<br> `<div role="button" aria-pressed="">Save</div>` while disallowing `<div role="button" aria-pressed>Save</div>`.

---
### Longer elaboration

As far as I can see, the wording *“Authors MAY specify a zero-length string ("") for any supported (but not required) state or property”* strictly disallows authors from having source markup like the following:

#### Example 1
```html
<div role="button" aria-pressed>Save</div>
```

In that example, the source markup hasn’t explicitly specified a zero-length string for the `aria-pressed` property — and therefore it doesn’t match that *“Authors MAY specify a zero-length string ("") for any supported (but not required) state or property”* requirement from the ARIA spec.

But to an HTML parser, the example above is identical to the following source markup:

#### Example 2
```html
<div role="button" aria-pressed="">Save</div>
```

That is, to an HTML parser, the source markup `aria-pressed` (with no value at all) and the source markup `aria-pressed=""` are exactly the same.

In other words, the source markup of the two examples above produce an identical DOM.

And the HTML checker doesn’t operate on source markup — instead it operates on the output produced by parsing the source markup with an HTML parser. So the HTML checker has no way to distinguish between **Example 1** and **Example 2** above.

Therefore, the only way I could implement the *“Authors MAY specify a zero-length string ("") for any supported (but not required) state or property”* requirement in the HTML checker is by assuming that **Example 1** above must be treated exactly the same as **Example 2** above.

More broadly, any tool that does conformance checking on the output of parsing the source markup with an HTML parser would need to make that same assumption in order to implement that requirement.
